### ISSUE TITLE
OOTB: Provide an option to globally disable the personal contact forms, and make it the default for new installations

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6067]
This feature request is a follow-up to this bug report: #6064

@klonos 
> ...I think that there should be a "Disable all personal contact forms" setting.

@jenlampton 
> I have always wanted this for contact module. And it should be disabled by default!

@kiamlaluno 
> I guess it would make sense in the case a site plans to use the Contact module just for the site contact form.

@olafgrabienski 
> Let's try to avoid "Disable to enable" situations. If it turns out that we need the setting, it should be "Enable personal contact forms".

**Comment by klonos** [comment]
There is no such setting in D7 (even after enabling the Contact module, which is disabled by default), but it exists in D9/10 OOTB (because the contact module is also enabled OOTB), under `admin/config/people/accounts`.
- The option/checkbox label is:
  > Enable the personal contact form by default for new users
- The description is:
  > Changing this setting will not affect existing users.

Here's what it looks like:
![image](https://user-images.githubusercontent.com/2423362/233081054-a0234101-a35b-4749-ba3c-8df5841f48e2.png)

**Comment by jenlampton** [comment]
> To me, it makes more sense to have a switch that is valid for all the user accounts, not just the new created ones.

+1 on this from me as well. 

We should also change the text on the checkbox to match the global enable / disable state.

**Comment by herbdool** [comment]
@avpaderno @jenlampton @olafgrabienski I've revived this and submitted a PR. There's a global setting for "Enable personal contact forms" which is enabled by default for backwards-compatibility on existing sites. And this is in addition to the existing setting, which I've moved to the contact settings page so it's all in the same place.

**Comment by izmeez** [comment]
The PR works for me. It appears that personal contact forms defaults to off, presumably for new sites and update hook sets to true on existing sites.

**Comment by herbdool** [comment]
@izmeez correct. The update hook ensures that it's backwards compatible with existing sites.

**Comment by izmeez** [comment]
Will this need a change record to alert existing sites so they can use the new setting? Or will it just need to be in the release notes.

**Comment by herbdool** [comment]
Normally we only make change records for code signature changes.

**Comment by avpaderno** [comment]
The PR works for me too. I just tested the live preview, not what happens when a site is updated.

**Comment by izmeez** [comment]
I have not tested a site update either.

**Comment by avpaderno** [comment]
Given that `contact_update_1003()` expressly checks for a `NULL` value, it only sets *personal* when it was not already set, as expected.

**Comment by quicksketch** [comment]
Thank you @herbdool, @avpaderno, and @izmeez for pulling this together! I merged https://github.com/backdrop/backdrop/pull/5321 into 1.x for 1.35.0. 🎉 Thank you @klonos, @jenlampton, and @olafgrabienski for your suggestions on how this should be built out.
